Nataly_Bondarenko/test_framework/tests/test_checkout_page.py

```diff
@@ -0,0 +1,97 @@
+        click_continue_button()
+    assert checkout_page.is_invalid_checkout_empty_values_error_message_displayed() is True, \
+        "Error message on empty values or empty username only, was not displayed"
+
+
+@pytest.mark.regression
+@pytest.mark.checkout
+@pytest.mark.parametrize("first_name_value, last_name_value, zip_value",
+                         [(f'{ReadConfig.get_first_name()}', "", ""),
+                          (f'{ReadConfig.get_first_name()}', "", f'{ReadConfig.get_zip()}')],
+                         ids=["first_name, empty last name, zip code",
+                              "first_name,  empty last name, empty zip code"])
+def test_invalid_checkout_empty_last_name(open_checkout_page, first_name_value, last_name_value, zip_value):
+    """
+    A method to verify error message if we try to go to the overview page from the checkout page:
+     1. if name and zip code fields in information box are empty
+     2. if lat name field is empty
+     Expected result: error msg with text: "Error: Last Name is required"
+    """
+    checkout_page = open_checkout_page
+    checkout_page.set_first_name(first_name_value).set_last_name(last_name_value).set_zip(zip_value). \
+        click_continue_button()
+    assert checkout_page.is_invalid_checkout_empty_last_name_error_message_displayed() is True, \
+        "Error message on empty last name value was not displayed"
+
+
+@pytest.mark.regression
+@pytest.mark.checkout
+def test_invalid_checkout_empty_zip(open_checkout_page):
+    """
+    A method to verify that we are not able to go to the overview page from the checkout page
+     if the zip code field is empty
+     Expected result: error msg with text: "Error: Postal Code is required"
+    """
+    checkout_page = open_checkout_page
+    checkout_page.set_first_name(f'{ReadConfig.get_first_name()}').set_last_name(f'{ReadConfig.get_last_name}') \
+        .set_zip("").click_continue_button()
+    assert checkout_page.is_invalid_checkout_empty_zip_error_message_displayed() is True, \
+        "Error message on empty zip value was not displayed"
+
+
+@pytest.mark.regression
+@pytest.mark.checkout
+def test_logout_from_checkout_page(open_checkout_page, open_login_page):
+    """
+    A method to verify that we are able to logout from the checkout page
+    """
+    checkout_page = open_checkout_page
+    checkout_page.click_menu_button().click_logout_button()
+    login_page = open_login_page
+    assert login_page.is_login_button_is_displayed() is True, "The logout button from checkout page doesn't work"
+
+
+@pytest.mark.regression
+@pytest.mark.checkout
+def test_go_to_cart_page_over_cart_link(open_checkout_page, open_cart_page):
+    """
+    A method to verify that we are able to go back to the cart page over cart link from the checkout page
+    """
+    checkout_page = open_checkout_page
+    checkout_page.click_cart_button()
+    cart_page = open_cart_page
```

Comment:
Why just not to return CartPage in click_cart_button()

Reply:
I  did it to avoid circular import.